src/services/transactions_queued.rs

```diff
@@ -43,26 +43,79 @@ pub fn get_queued_transactions(
         .request_cached(&context.client(), &url, request_cache_duration())?;
     let mut backend_transactions: Page<MultisigTransaction> = serde_json::from_str(&body)?;
 
+    // We need to do this before we create the iterator
+    // Nonce of the first item in the next page (-1 if not present)
+    let edge_nonce = get_edge_nonce(&mut backend_transactions);
+
+    // Use an iterator to avoid shifting the result vector (would potentially trigger copies)
+    let mut tx_iter = backend_transactions.results.into_iter();
+    // Nonce of the last item in the previous page (-1 if not present)
+    let previous_page_nonce = get_previous_page_nonce(&page_meta, &mut tx_iter);
+
+    let service_transactions = process_transactions(
+        &mut info_provider,
+        safe_nonce,
+        &mut tx_iter,
+        previous_page_nonce,
+        edge_nonce,
+    );
+
+    Ok(Page {
+        next: build_page_url(
+            context,
+            &safe_address,
+            &page_meta,
+            timezone_offset,
+            display_trusted_only,
+            backend_transactions.next,
+            1, // Direction forward
+        ),
+        previous: build_page_url(
+            context,
+            &safe_address,
+            &page_meta,
+            timezone_offset,
+            display_trusted_only,
+            backend_transactions.previous,
+            -1, // Direction backwards
+        ),
+        results: service_transactions,
+    })
+}
+
+// Nonce of the first item in the next page (-1 if not present)
+fn get_edge_nonce(backend_transactions: &mut Page<MultisigTransaction>) -> i64 {
     // If there is a next url we remove the last item for information on the next page
-    let after_tx = if backend_transactions.next.is_some() {
+    if backend_transactions.next.is_some() {
         backend_transactions.results.pop()
     } else {
         None
-    };
-    // Use an iterator to avoid shifting the result vector (would potentially trigger copies)
-    let mut tx_iter = backend_transactions.results.into_iter();
+    }
+    .map_or(-1, |tx| tx.nonce as i64)
+}
+
+// Nonce of the last item in the previous page (-1 if not present)
+fn get_previous_page_nonce(
+    page_meta: &PageMetadata,
+    tx_iter: &mut dyn Iterator<Item = MultisigTransaction>,
+) -> i64 {
     // If we are not on the first page then we take the first item to get information on the previous page
-    let before_tx = if page_meta.offset == 0 {
+    if page_meta.offset == 0 {
         None
     } else {
         tx_iter.next()
-    };
-    // Nonce of the first item in the next page (-1 if not present)
-    let edge_nonce = after_tx.map_or(-1, |tx| tx.nonce as i64);
-    // Nonce of the last item in the previous page (-1 if not present)
-    let previous_page_nonce = before_tx.map_or(-1, |tx| tx.nonce as i64);
-    let mut last_proccessed_nonce = previous_page_nonce;
+    }
+    .map_or(-1, |tx| tx.nonce as i64)
+}
 
+fn process_transactions(
```

Comment:
Generally for me, `service_transactions` correspond to the types from the gateway as opposed to the `backend_transaction` and in this case `tx_list_item` as this refers to a screen on the app rather than a domain type. If you agree with this naming, I would change the name of `service_transactions` to `tx_list_items`. `tx_iter` is fine though, as the `MultisigTransaction` type is right next to it and then we know those are backend txs.

Reply:
no strong preferences here